sagemaker/pipeline/create_sm_pipeline.py

```diff
@@ -0,0 +1,693 @@
+            "GPUInstanceType", args.instance_config.gpu_instance_type
+        )
+        self.graphconstruct_instance_type_param = self._create_string_parameter(
+            "GraphConstructInstanceType",
+            args.instance_config.graph_construction_instance_type,
+        )
+        self.volume_size_gb_param = self._create_int_parameter(
+            "InstanceVolumeSizeGB",
+            args.instance_config.volume_size_gb,
+        )
+        self.execution_subpath_param = self._create_string_parameter(
+            "ExecutionSubpath", args.get_hash_hex()
+        )
+        self.graphconstruct_config_param = self._create_string_parameter(
+            "GraphConstructConfigFile", args.graph_construction_config.config_filename
+        )
+        self.partition_algorithm_param = self._create_string_parameter(
+            "PartitionAlgorithm", args.partition_config.partition_algorithm
+        )
+        # TODO: Probably should not be a parameter
+        self.graph_name_param = self._create_string_parameter(
+            "GraphName", args.task_config.graph_name
+        )
+        self.num_trainers_param = self._create_int_parameter(
+            "NumTrainers", args.training_config.num_trainers
+        )
+        # TODO: Maybe should not be configurable because it requires changing job sequence?
+        self.use_graphbolt_param = self._create_string_parameter(
+            "UseGraphBolt", args.training_config.use_graphbolt_str
+        )
+        self.input_data_param = self._create_string_parameter(
+            "InputData", args.task_config.input_data_s3
+        )
+        self.output_prefix_param = self._create_string_parameter(
+            "OutputPrefix", args.task_config.output_prefix
+        )
+        self.train_config_file_param = self._create_string_parameter(
+            "TrainConfigFile", args.training_config.train_yaml_file
+        )
+
+        # If inference yaml is not provided, re-use the training one
+        inference_yaml_default = (
+            args.inference_config.inference_yaml_file
+            or args.training_config.train_yaml_file
+        )
+        self.inference_config_file_param = self._create_string_parameter(
+            "InferenceConfigFile", inference_yaml_default
+        )
+        self.inference_model_snapshot_param = self._create_string_parameter(
+            "InferenceModelSnapshot", args.inference_config.inference_model_snapshot
+        )
+
+    def _create_pipeline_steps(
+        self, args: PipelineArgs
+    ) -> List[Union[ProcessingStep, TrainingStep]]:
+        """Creates the pipeline steps and reeturns them as a list
+
+        Parameters
+        ----------
+        args : PipelineArgs
+            Pipeline arguments
+
+        Returns
+        -------
+        List[Union[ProcessingStep, TrainingStep]]
+            A list of pipeline steps to be executed in sequence
+        """
+        pipeline_steps: List[Union[ProcessingStep, TrainingStep]] = []
```

Comment:
What about inference step?

Reply:
These are SageMaker Pipeline classes, see https://docs.aws.amazon.com/sagemaker/latest/dg/build-and-manage-steps.html#build-and-manage-properties.

For inference we still use a SageMaker Training job.

Reply:
Can you add a comment here, saying Inference will also run as a SageMaker Training job.